pyroapi/dispatch.py

```diff
@@ -50,7 +50,8 @@ def pyro_backend(*aliases, **new_backends):
         old_backends[name] = GenericModule.current_backend[name]
         GenericModule.current_backend[name] = new_backend
     try:
-        yield
+        with handlers.seed(rng_seed=1):
```

Comment:
Hmm..one big issue that I see with this is that this will make it impossible to reset the seed to anything other than 1, at least in NumPyro.

i.e. 
```python
with handlers.seed(rng_seed=1):
  with handlers.seed(rng_seed=2):
    pyro.sample('x', ...)  # only the outermost effect handler dictates the seeding in NumPyro
```

This is because `apply_stack` is applied in reverse order. This is different from my implementation of the seed handler in Pyro which sets the seed in `__enter__` and therefore the inner one dictates the seeding in Pyro. What do you think about changing the effect handler implementation in NumPyro so that the behavior matches that of Pyro, i.e. the innermost context dictates the seeding? Then we should be able to use this default seeding which is overrideable by the tests.

Reply:
I don't know your details in NumPyro, but innermost seeding sounds right to me.